### PR TITLE
Make OW4 ready for aws

### PR DESCRIPTION
**Pull request by tokongs**
## What kind of a pull request is this?

- QA / Code Review

## Code Checklist

- [ ] The code follows dotkom code style 
- [ ] The code passes the defined tests
- [ ] I have added tests for the code I added
- [ ] I have provided documentation for the code I added
- [ ] The code is ready to be merged

## (Possible) Breaking Changes

- [ ] The changes are backwards compatible

- [x] I have updated the build configuration
- [x] These changes requires changes to configuration in production 
    - [ ] I have applied the required changes in production
    - [ ] I cannot apply the required changes in production before this is deployed.

## Description of changes
- Change settings so that the ones we care about
can be set as env variables.
- Add AWS settings file with aws specific settings.
- Use redis as cache instead of memcached when using AWS.
- Create a basic health check middleware.
- Create two storage backend based on boto3 s3 backend.
- Change offline app to wait for transaction before starting task.
- Change gallary app to work with s3.
- Change how we authenticate with gsuite.

### CODE REVIEW

File: Dockerfile.prod [hunk 1]
```diff
@@ -24,21 +24,17 @@ RUN apt-get update && apt-get install -y \
   git \
   gcc \
   curl \
-  mime-support
+  mime-support && \
+  pip install poetry
```
[the remark this answers is not in the document]
Comment (tokongs) [reply Dockerfile.prod:28]: Combining these makes the build go a little slower some(very few)times, but results in a smaller image. As far as I understand...

File: apps/gsuite/auth.py [hunk 2]
```diff
@@ -2,23 +2,31 @@
...
+    try:
+        account = ServiceAccount.objects.all()[:1]
+        info = serializers.serialize("json", account)
+        info = json.loads(info)
+        info = info[0]["fields"]
+    except:
+        info = ""
+
+    credentials = service_account.Credentials.from_service_account_info(
+        info, scopes=scopes
```
[the remark this answers is not in the document]
Comment (tokongs) [reply apps/gsuite/auth.py:29]: To have it as a file works very poorly when you don't have a persistent file system. I tried to make it work with a custom storage backend for s3, like I have done with media, but I just couldn't get it to work. If anyone wants to try, feel free. Doing it with env variables should be possible though.

File: apps/offline/signals.py [hunk 3]
```diff
@@ -12,4 +15,5 @@ def trigger_create_thumbnail(sender, instance: Issue, created=False, **kwargs):
     Calls the create thumbnail task if an issue saved (either created or updated).
     """
     if not instance.image:
-        create_thumbnail_task.delay(issue_id=instance.id)
+        transaction.on_commit(lambda: create_thumbnail_task.delay(issue_id=instance.id))
```
Comment (oleast) [inline apps/offline/signals.py:18]: Why is this needed?

Comment (tokongs) [reply apps/offline/signals.py:18]: I experienced a race condition where the celery task executed before the dB transaction committed when running on AWS. So when the task tried to get the object it couldn't because it wasn't in the dB yet. This resolves that bh waiting for the transaction to be committed.
